File: frontend/lib/shared/models/lyrics.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'lyrics.freezed.dart';
part 'lyrics.g.dart';

@freezed
sealed class Lyrics with _$Lyrics {
  const factory({
    required int id,
    @JsonKey(name: 'media_id') required int mediaId,
    required String source,
    @JsonKey(name: 'created_at') required DateTime createdAt,
    @JsonKey(name: 'updated_at') required DateTime updatedAt,
    @JsonKey(name: 'lyrics_text') @Default('') String lyricsText,
    @JsonKey(name: 'translation') @Default('') String translation,
    @JsonKey(name: 'sync_data') @Default('') String syncData,
  }) = _Lyrics;

  factory fromJson(Map<String, dynamic> json) => _$LyricsFromJson(json);
}

File: backend/internal/models/lyrics.go
package models

import "time"

// Lyrics holds optional lyrics text, translation, and sync timestamps for a media item.
type Lyrics struct {
	ID          uint   `gorm:"primaryKey" json:"id"`
	MediaID     uint   `gorm:"uniqueIndex" json:"media_id"`
	LyricsText  string `gorm:"type:text" json:"lyrics_text"`
	Translation string `gorm:"type:text" json:"translation"`
	SyncData    string `gorm:"type:text" json:"sync_data"` // JSON array of {timestamp, text} pairs
	Source      string `gorm:"size:255" json:"source"`     // id3, manual, external
	// FK-ассоциация с каскадным удалением (для новых инсталляций).
	Media     *Media    `gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE" json:"-"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}
